t/issue/1508.t

```diff
@@ -0,0 +1,165 @@
+
+=head1 NAME 
+
+issue/1508.t  - Check that the `needs()` function correctly propogates run-time task arguments
+
+=head1 DESCRIPTION
+
+Check that the `needs()` function does indeed correctly propogate 
+run-time task arguments (%params and @args) from the calling task down to the "needed" tasks.
+
+=head1 DETAILS
+
+  * AUTHOR / DATE : [tabulon]@[2021-09-26]
+  * RELATES-TO    : [github issue #1508](https://github.com/RexOps/Rex/issues/1508#issue-1007457392)
+  * INSPIRED from : t/needs.t
+
+=cut
+
+use Test::More;
+use Rex::Commands;
+
+{
+
+  package T; # Helper package (for cutting down boilerplate in tests)
+  use Storable;
+
+  sub track_taskrun {
+    my %opts = ref $_[-1] eq 'HASH'
+      ? %{
+      ;
+      pop
+      }
+      : ();
+    my $argv    = $opts{argv};
+    my @prereqs = (@_);
+    for (@prereqs) {
+      my $file = "${_}.txt";
+      Storable::store( $argv, $file );
+    }
+  }
+
+  sub check_needed {
+    my %opts = ref $_[-1] eq 'HASH'
+      ? %{
+      ;
+      pop
+      }
+      : ();
+    my $argv      = $opts{argv};
+    my $do_unlink = delete $opts{unlink} // 1;
+    my @prereqs   = (@_);
+
+    for (@prereqs) {
+      my $file = "${_}.txt";
+
+      -f "$file" or die;
+      my $propagated_argv = Storable::retrieve("$file");
+      Test::More::_deep_check( $propagated_argv, $argv ) or die;
```

Comment:
What are the advantages of using the private `_deep_check` explicitly here instead of the public `is_deeply`?

Reply:
Replied elsewhere